src/transformers/models/patchtst/modeling_patchtst.py

```diff
@@ -0,0 +1,2091 @@
+        else:
+            ## Position-wise Feed-Forward and Add residual connection and Norm
+            src = self.norm_sublayer3(
+                src + self.dropout_path3(self.ff(src))
+            )  # Add: residual connection with residual dropout
+        src = src.reshape(
+            batch_size, num_input_channels, sequence_length, d_model
+        )  # [bs x num_channels x sequence_length x d_model]
+
+        return src
+
+
+class PatchTSTPreTrainedModel(PreTrainedModel):
+    config_class = PatchTSTConfig
+    base_model_prefix = "model"
+    main_input_name = "past_values"
+    supports_gradient_checkpointing = False
+
+    def _init_weights(self, module):
+        """Initialize weights"""
+        if self.config.use_cls_token:
+            nn.init.normal_(self.config.cls_token, std=0.02)
+        elif isinstance(module, nn.LayerNorm):
+            module.bias.data.zero_()
+            module.weight.data.fill_(1.0)
+        elif isinstance(module, (nn.Linear, nn.Conv1d)):
+            module.weight.data.normal_(mean=0.0, std=self.config.init_std)
+            if module.bias is not None:
+                module.bias.data.zero_()
+
+    def _set_gradient_checkpointing(self, module, value=False):
+        if isinstance(module, (PatchTSTEncoder)):
+            module.gradient_checkpointing = value
+
+
+class PatchTSTEncoder(PatchTSTPreTrainedModel):
+    """
+    PatchTST Encoder
+    """
+
+    def __init__(self, config: PatchTSTConfig):
+        super().__init__(config)
+        self.num_input_channels = config.num_input_channels
+        self.num_patches = config.num_patches
+        self.patch_length = config.patch_length
+        self.d_model = config.d_model
+        self.shared_embedding = config.shared_embedding
+        self.use_cls_token = config.use_cls_token
+        self.gradient_checkpointing = False
+
+        # Input encoding: projection of feature vectors onto a d-dim vector space
+        if not config.shared_embedding:
+            self.input_embedding = nn.ModuleList()
+            for _ in range(self.num_input_channels):
+                self.input_embedding.append(nn.Linear(config.patch_length, config.d_model))
+        else:
+            self.input_embedding = nn.Linear(config.patch_length, config.d_model)
+
+        # Positional encoding
+        if config.use_cls_token:
+            self.cls_token = nn.Parameter(torch.zeros(1, 1, 1, config.d_model))
+            self.position_enc = positional_encoding(
+                config.positional_encoding, config.learn_pe, config.num_patches + 1, config.d_model
+            )
+        else:
+            self.position_enc = positional_encoding(
+                config.positional_encoding, config.learn_pe, config.num_patches, config.d_model
+            )
+
+        # Positional dropout
+        self.positional_dropout = (
+            nn.Dropout(config.positional_dropout) if config.positional_dropout > 0 else nn.Identity()
+        )
```

Comment:
this part can be encapsulated in a `PatchTSTEmbedding` class. Then we only need a single `PatchTSTEncoder` class, which consists of an `nn.ModuleList` of `PatchTSTEncoderLayer` objects.

Reply:
Yes, I agree. PatchTSTEncoderBlock class is removed and we only need a single PatchTSTEncoder class.

Reply:
Thank you! It has been resolved.